crates/dojo-bindgen/src/plugins/recs/mod.rs

```diff
@@ -0,0 +1,618 @@
+        for model in models {
+            let tokens = &model.tokens;
+
+            let mut sorted_structs = tokens.structs.clone();
+            sorted_structs.sort_by(compare_tokens_by_type_name);
+
+            let mut sorted_enums = tokens.enums.clone();
+            sorted_enums.sort_by(compare_tokens_by_type_name);
+
+            for token in &sorted_enums {
+                handled_tokens.push(token.to_composite().unwrap().to_owned());
+            }
+            for token in &sorted_structs {
+                handled_tokens.push(token.to_composite().unwrap().to_owned());
+            }
+
+            for token in &sorted_enums {
+                if handled_tokens.iter().filter(|t| t.type_name() == token.type_name()).count() > 1
+                {
+                    continue;
+                }
+                out += TypescriptRecsPlugin::format_enum(token.to_composite().unwrap()).as_str();
+            }
+
+            for token in &sorted_structs {
+                if handled_tokens.iter().filter(|t| t.type_name() == token.type_name()).count() > 1
+                {
+                    continue;
+                }
+
+                // first index is our model struct
+                if token.type_name() == naming::get_name_from_tag(&model.tag) {
+                    models_structs.push((
+                        naming::get_namespace_from_tag(&model.tag),
+                        token.to_composite().unwrap().clone(),
+                    ));
+                }
+
+                out += TypescriptRecsPlugin::format_struct(token.to_composite().unwrap()).as_str();
+            }
+
+            out += "\n";
+        }
+
+        out += "
+export function defineContractComponents(world: World) {
+    return {
+";
+
+        for (namespace, model) in models_structs {
+            out += TypescriptRecsPlugin::format_model(&namespace, &model).as_str();
+        }
+
+        out += "    };
+}\n";
+
+        out
+    }
+
+    // Formats a system into a C# method used by the contract class
+    // Handled tokens should be a list of all structs and enums used by the contract
+    // Such as a set of referenced tokens from a model
+    fn format_system(system: &Function, handled_tokens: &[Composite], namespace: String) -> String {
+        if [
+            "contract_name",
+            "namespace",
+            "tag",
+            "name_hash",
+            "selector",
+            "dojo_init",
+            "namespace_hash",
+        ]
+        .contains(&system.name.as_str())
+        {
+            return String::new();
+        }
+        fn map_type(token: &Token) -> String {
+            match token {
+                Token::CoreBasic(_) => TypescriptRecsPlugin::map_type(token)
+                .replace("RecsType.", "").replace("Array", "[]")
+                // types should be lowercased
+                .to_lowercase(),
+                Token::Composite(t) => format!("models.{}", t.type_name()),
```

Comment:
Are you sure at this level the composites tokens are only models?
You may end up with some enums that are simply types passed as inputs to systems?

Could be interesting to test with some enums and structs the are not models. Or models is a generic module where you have all the types?